core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -316,17 +323,16 @@ object AdminUtils extends Logging with AdminUtilities {
     var partitionId = startPartitionId
     partitionList = partitionList.takeRight(partitionList.size - partitionId)
     for (i <- partitionList.indices) {
-      val brokerList = partitionList(i).split(":").map(s => s.trim().toInt)
+      val brokerList = partitionList(i).split(":").map(s => s.trim().toInt).toList
       if (brokerList.isEmpty)
-        throw new AdminOperationException("replication factor must be larger than 0")
+        throw new InvalidReplicaAssignmentException("Replication factor must be larger than 0.")
       if (brokerList.size != brokerList.toSet.size)
-        throw new AdminOperationException("duplicate brokers in replica assignment: " + brokerList)
+        throw new InvalidReplicaAssignmentException(s"Duplicate brokers in replica assignment: ${brokerList.mkString(", ")}.")
       if (checkBrokerAvailable && !brokerList.toSet.subsetOf(availableBrokerList))
-        throw new AdminOperationException("some specified brokers not available. specified brokers: " + brokerList +
-          "available broker:" + availableBrokerList)
-      ret.put(partitionId, brokerList.toList)
-      if (ret(partitionId).size != ret(startPartitionId).size)
-        throw new AdminOperationException("partition " + i + " has different replication factor: " + brokerList)
+        throw new AdminOperationException(s"Some specified brokers not available. Specified brokers: ${brokerList.mkString(", ")}, available brokers: ${availableBrokerList.mkString(", ")}.")
+      ret.put(partitionId, brokerList)
+      if (brokerList.size != ret(startPartitionId).size)
+        throw new InvalidReplicaAssignmentException(s"Partition ${i+startPartitionId} has different replication factor: ${brokerList.mkString(", ")}.")
```

Comment:
We should probably have a separate method for validating this and provide a more useful message. Maybe something along the lines of "Inconsistent replication factor between partitions, partition 0 has X while partitions a, b, c have different values". Seems like we're doing something like this in `addPartitions`.